File: module4/casestudy/v1/src/main/java/com/example/v1/entity/customer/Customer.java
package com.example.v1.entity.customer;

import com.example.v1.entity.contract.Contract;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.*;

import javax.persistence.*;
import java.util.Set;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    private String dateOfBirth;
    private Integer gender;
    private String idCard;
    private String phoneNumber;
    private String email;
    private String address;
    private Integer isDelete = 1;

    @ManyToOne
    @JoinColumn(name = "customer_type_id", referencedColumnName = "id")
    private CustomerType customerType;

    @OneToMany(mappedBy = "customer")
    private Set<Contract> contracts;
}

File: module5/furama/src/app/model/customer/customer.ts
import {CustomerType} from './customer-type';
import {Gender} from '../gender';

export interface Customer {
  id: number;
  name: string;
  dateOfBirth?: Date;
  gender: Gender;
  idCard: number;
  phoneNumber: number;
  email: string;
  address: string;
  customerType: CustomerType;

}
